backend/app/views/spree/admin/taxons/_form.html.erb

```diff
@@ -20,7 +20,20 @@
       </div>
     <% end %>
 
-    <%= render "spree/admin/taxons/attachment_forms/#{f.object.attachment_partial_name}", f: f %>
+    <%= f.field_container :icon do %>
```

Comment:
IMO It still needs to iterate over the attachment definitions, it is pretty common to add more than icon attachment to taxons, hero image, mobile hero image, brand logo, etc

Reply:
I'm not against that in principle, I just thought that if done it should have its own documentation/guide with complete support. 

On the same note might make sense to completely remove the icon from the taxon and let individual store add it if needed, allowing to manage the whole attachment configuration (e.g. pick sizes, attach a video instead, etc.).

In my experience it's always the case that a store has it's own rules for images and attachments.

@softr8 would any of these alternative work for you?

Reply:
hi folks

I'm jumping in as we've added a lot of different attachments to our taxon object and do not use the default icon.
So as @elia, I'm in favour of removing the icon from the core model. And maybe move it into an extension if some store use only that icon, but my guess is that it won't be used that much.

Reply:
I'm open to someone trying to change my mind, but it feels like removing support for multiple taxon attachments, even if not documented/explicitly supported, would effectively constitute a breaking change.

As soon as we release some kind of public API, we _must_ assume that people are using it and we can't just pull the rug from under their feet, no matter how well documented it is, or whether it was introduced intentionally or by mistake.

With that said, I agree that we should probably remove the default "icon" attachment altogether and simply let people define their own attachments — but I also think that should be considered a breaking change.

Curious to hear if anyone from @solidusio/core-team feels differently about this.

Reply:
I added the usual deprecation/configuration harness 👍